### ISSUE TITLE
ParseWithMetaLookups produces wrong projection on self join

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11132]
This test fails

```java
@Test
public void testJoinWithAliasedTables() {
    withMeta(U);

    Unquoted T1 = U.as("T1");
    Unquoted T2 = U.as("T2");

    assertEquals(asList(T1.COL1, T2.COL1),
        dsl.parser().parseSelect("select T1.COL1, T2.COL1 from U as T1 join U as T2 on T1.COL1 = T2.COL1").getSelect());
}
```

Error:

```
java.lang.AssertionError: expected:<["T1".COL1, "T2".COL1]> but was:<
  T1.COL1,
  T1.COL1>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.jooq.test.util.BaseTest.assertEquals(BaseTest.java:122)
	at org.jooq.impl.ParserMetaTest.testJoinWithAliasedTables(ParserMetaTest.java:337)
```

There's also a warning in the logs:

```
11:30:24,618  INFO [org.jooq.impl.FieldsImpl                          ] - Ambiguous match found for COL1. Both T1.COL1 and T2.COL1 match.
java.sql.SQLWarning: null
	at org.jooq.impl.FieldsImpl.field0(FieldsImpl.java:194) [classes/:?]
	at org.jooq.impl.FieldsImpl.field(FieldsImpl.java:172) [classes/:?]
	at org.jooq.impl.AbstractRow.field(AbstractRow.java:170) [classes/:?]
	at org.jooq.impl.AbstractTable.field(AbstractTable.java:240) [classes/:?]
	at org.jooq.impl.ParserContext.scopeEnd(ParserImpl.java:12549) [classes/:?]
	at org.jooq.impl.ParserContext.parseSelect(ParserImpl.java:1287) [classes/:?]
	at org.jooq.impl.ParserContext.parseSelect(ParserImpl.java:1145) [classes/:?]
	at org.jooq.impl.ParserContext.parseQuery(ParserImpl.java:1002) [classes/:?]
	at org.jooq.impl.ParserContext.parseSelect0(ParserImpl.java:805) [classes/:?]
	at org.jooq.impl.ParserImpl.parseSelect(ParserImpl.java:662) [classes/:?]
	at org.jooq.impl.ParserImpl.parseSelect(ParserImpl.java:657) [classes/:?]
	at org.jooq.impl.ParserMetaTest.testJoinWithAliasedTables(ParserMetaTest.java:338) [test-classes/:?]
```

**Comment by lukaseder** [comment]
The `org.jooq.impl.ParserContext.lookupFields` variable is of type `ScopeStack<String, FieldProxy<?>>` when it should be a `ScopeStack<Name, FieldProxy<?>>`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11142)
